core/modules/macros/contrastcolour.js

```diff
@@ -29,6 +29,22 @@ exports.params = [
 Run the macro
 */
 exports.run = function(target,fallbackTarget,colourA,colourB) {
```

Comment:
> I'm afraid I don't see the purpose of these changes to the contrastcolour macro. Can you describe the change you're making more clearly?

Compare the existing wikitext construct passing parameters from one superfluous helper macro to the next.

All the parameters that are now being passed to the macro are intrinsic to the title of the tiddler we're at with the TagTemplate. The colorcontrast macro shouldn't be so special as to only know of colors but not tiddlers.

However, contrastcolor is but one change I made.

The other changes fix a broken edit template and harmonize using and reusing the tag macro in both view and edit mode by a more modular structure rather than repetitive and quite bloated templates.